### ISSUE TITLE
mock function fails for golang lambdas with custom return type

### ISSUE THREAD

**Issue by jnathanh** [aws-amplify/amplify-cli#4487]
**Describe the bug**
When using a golang function with a custom return type, mocking the function locally returns the following error:

>Error: Unknown error occurred during the execution of the Lambda function
    at Object.exports.localInvoke (/usr/local/Cellar/node/13.12.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-go-function-runtime-provider/lib/localinvoke.js:78:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)

**Amplify CLI Version**
4.21.0

**To Reproduce**
Create golang function
```shell
$ amplify add function helloworld
Using service: Lambda, provided by: awscloudformation
? Provide a friendly name for your resource to be used as a label for this category in the project: helloworld
? Provide the AWS Lambda function name: helloworld
? Choose the function runtime that you want to use: Go
Only one template found - using Hello World by default.
? Do you want to access other resources created in this project from your Lambda function? No
? Do you want to invoke this function on a recurring schedule? No
? Do you want to edit the local lambda function now? No
```
Replace the function main.go with the following code (custom return type handler example [from aws docs](https://docs.aws.amazon.com/lambda/latest/dg/golang-handler.html#golang-handler-structs))
```go
package main

import (
  "fmt"
  "github.com/aws/aws-lambda-go/lambda"
)

type MyEvent struct {
  Name string `json:"What is your name?"`
  Age int     `json:"How old are you?"`
}

type MyResponse struct {
  Message string `json:"Answer:"`
}

func HandleLambdaEvent(event MyEvent) (MyResponse, error) {
  return MyResponse{Message: fmt.Sprintf("%s is %d years old!", event.Name, event.Age)}, nil
}

func main() {
  lambda.Start(HandleLambdaEvent)
}
```
Replace the sample event.json file with the following json
```json
{
  "What is your name?": "Jim",
  "How old are you?": 33
}
```
Run the mock function command
```shell
$ amplify mock function helloworld
? Provide the path to the event JSON object relative to /Users/nhanna/code/threes/amplify/backend/function/helloworld src/event.json
Starting execution...
Launching Lambda process, port: 8000
helloworld failed with the following error:
Error: Unknown error occurred during the execution of the Lambda function
    at Object.exports.localInvoke (/usr/local/Cellar/node/13.12.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-go-function-runtime-provider/lib/localinvoke.js:78:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Finished execution.
```

**Expected behavior**
No Errors and a valid response output to stderr. The error output should also be more actionable in the case that this might be user error.

**Desktop (please complete the following information):**
 - OS: macOS Catalina (10.15.4)
 - Node Version: v10.16.3
- Go Version: go1.14.3 darwin/amd64

**Additional context**
You can tell this is an error in the type handling because if you manually serialize to json and return as a string it works just fine.

**Comment by zerox92** [comment]
I created a deployment package following doc [here](https://docs.aws.amazon.com/lambda/latest/dg/golang-package.html) using the same handler returning struct type and it was working fine for me. The issue I guess is with 'amplify mock'.

![Lambda_Invoke](https://user-images.githubusercontent.com/25776125/84526466-40006280-acd5-11ea-9ab4-f974d4636c61.png)

**Comment by jqdurham** [comment]
I just ran into this as well.  In our case, it's a Lambda invoked via API Gateway with the following signature:

`func HandleRequest(ctx context.Context, event events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error)`

**Comment by TrungRueta** [comment]
Confirm 
**v4.24.2**

```go
package main

import (
	"context"

	"github.com/aws/aws-lambda-go/lambda"
)

type MyEvent struct {
	Name string `json:"name"`
}

func HandleRequest(ctx context.Context, input MyEvent) (MyEvent, error) {
	return input, nil
}

func main() {
	lambda.Start(HandleRequest)
}
```

```json
{
  "name": "Amplify"
}
```

run : `amplify mock function <name function>`

ouput:
```Starting execution...
Local invoker binary was not found, building it...
Launching Lambda process, port: 8000
staticInfo failed with the following error:
Error: Unknown error occurred during the execution of the Lambda function
    at Object.exports.localInvoke (/Users/trungrueta/.nvm/versions/node/v11.15.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-go-function-runtime-provider/lib/localinvoke.js:79:19)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)```

Look like Go function mock only work when return string.

**Comment by zerox92** [comment]
A fix is up for this: https://github.com/aws-amplify/amplify-cli/pull/4915
PR is currently in review

**Comment by SwaySway** [comment]
Closing as this fix for this has been merged and will be included in the next release (after v4.24.3)
